README.md

```diff
@@ -1,3 +1,67 @@
+# Knowledge Graph Pipeline
+
+The Knowledge Graph will help builders of data platforms to find relevant datasets.
+
+This repository is a proof of concept of a data pipeline for generating such a Knowledge Graph.
+
+## Approach
+
+The Knowledge Graph is built on top of
+the [Dataset Register](https://github.com/netwerk-digitaal-erfgoed/dataset-register), which contains dataset
+descriptions as supplied by their owners. Part of these descriptions are distributions, i.e. URLs where the data can be
+retrieved.
+
+The Knowledge Graph loads RDF data from the distributions and applies SPARQL queries to it to build **dataset
+summaries**. Summaries answer questions such as:
+
+- which RDF types are used in the dataset?
+- for each of those types, how many resources does the dataset contain?
+- which predicates are used in the dataset?
+- which URI prefixes to the data link to?
+- for each of those prefixes, which match known [terminology sources](https://termennetwerk.netwerkdigitaalerfgoed.nl)?
+- for each of those sources, how many outgoing links to them does the dataset have?
+- (and more)
+
+The summaries can be consulted by users such as data platform builders to help them find relevant datasets.
+
+## Scope
+
+- This pipeline is RDF-based so will be limited to datasets that provide at least one valid RDF distribution.
+
+## Pipeline Steps
+
+The pipeline will consist of the following steps.
+
+### 1. Retrieve datasets with RDF distributions from the Dataset Register.
+
+The Knowledge Graph is limited in scope to RDF distributions.
+
+### 2. Load RDF data from the distribution.
+
+This can be either in-memory or in a triple store. The latter may perform better because we need to apply SPARQL queries
+to the RDF.
+
+### 3. Apply analysis queries to the loaded data.
+
+There are two sets of queries:
+
+- generic queries that apply to all datasets (for example all types);
+- dataset-specific queries for extracting detailed information.
+
+The SPARQL queries will be stored in this Git repository, in a predetermined directory structure. This way, no
+configuration will be necessary.
+
+All queries are SPARQL `CONSTRUCT` queries that output analysis results as triples (for example
+in [VoID](https://www.w3.org/TR/void/)).
+
+For now the queries will be self-contained, complete units. In the future, we may want to combine queries with extra
+code (TypeScript) functions for extracting even more detailed information, such as image resolution.
+
+### 4. Write analysis results
+
+Write the results of the analysis queries to local files. The results may also be inserted into a triple store that can
+then be consulted by clients as a Knowledge Graph.
```

Comment:
As this a proof of concept, several functions (like data import) and queries as timed. These timings are part of the analysis.